config/config.go

```diff
@@ -129,7 +129,7 @@ func (c WalkParams) ConfigureSNMP(g *gosnmp.GoSNMP) {
 		g.Version = gosnmp.Version3
 	}
 	g.Community = string(c.Auth.Community)
-	g.ContextName = string(c.Auth.ContextName)
+	g.ContextName = c.Auth.ContextName
```

Comment:
I wonder why this one, and not the `c.Auth.Community` above?

Reply:
@SuperQ because `c.Auth.Community` is typed `Secret`.

```golang
// Secret is a string that must not be revealed on marshaling.
type Secret string
[...]
type Auth struct {                                                              
        Community     Secret `yaml:"community,omitempty"`
````

Reply:
Ahh, right! Thanks!